include/irr/asset/ICPUBuffer.h

```diff
@@ -12,13 +12,14 @@
 #include "irr/core/alloc/null_allocator.h"
 #include "irr/asset/IAsset.h"
 #include "irr/asset/bawformat/blobs/RawBufferBlob.h"
+#include "irr/asset/IDescriptor.h"
 
 namespace irr
 {
 namespace asset
 {
 
-class ICPUBuffer : public core::IBuffer, public asset::IAsset
+class ICPUBuffer : public core::IBuffer, public asset::IAsset, public IDescriptor

```

Comment:
yeah, about that chain of inheritance

Reply:
@Crisspl ?

Reply:
is it question? idk what i can say here

Reply:
same issue as here
https://github.com/buildaworldnet/IrrlichtBAW/pull/358#discussion_r328329575

Reply:
@Crisspl TODO.

Reply:
resolved

Reply:
core::IBuffer needs to stay in `core` its used for low level refcounted memory managment.

ICPU and IGPU Buffer need a common base on top of core::IBuffer not moving of IBuffer our of core.

Reply:
resolved